### PR TITLE
Fix: Type Updates

### PR DESCRIPTION
**Pull request by nrayburn-tech**
## What kind of change does this PR introduce?

Type updates/bug fix

## What is the current behavior?

Not type safe through the use of `any`.

## What is the new behavior?

Type safe with more specific types.

## Additional context

Mostly just type updates, but some components had a bit more work done for cleanup after the types were made more explicit.  (Remove unnecessary optional chaining,`Number` calls, etc.)

Changes were broken up by commit, so viewing one commit at a time will be easier to review.

Ref #6249.  Doesn't close the issue, but is a start.

### CODE REVIEW

**Comment by Isaiah-Hamilton** [comment]
@nrayburn-tech Thank you for the pull request, but next time can you lets us know that you want to pick up the issue. Especially if its assigned. Still greatly appreciated and will review.

**Comment by nrayburn-tech** [comment]
Sorry about that.

This PR doesn’t completely close the issue, but it is a starting point. If things look good, I can submit some more PRs to keep cleaning up the folder.

File: studio/components/to-be-cleaned/TooltipListener.tsx [hunk 1]
```diff
@@ -1,80 +0,0 @@
-import { useRef, useEffect } from 'react'
```
Comment (Isaiah-Hamilton) [inline studio/components/to-be-cleaned/TooltipListener.tsx:1]: Why is this file being removed?

**Comment by Isaiah-Hamilton** [comment]
@nrayburn-tech Sorry for the late review.

File: studio/components/to-be-cleaned/TooltipListener.tsx (hunk 1 above)
Comment (nrayburn-tech) [reply studio/components/to-be-cleaned/TooltipListener.tsx:1]: It's only usage is in the `deprecated.ts` file to re-export it, but it isn't used anywhere outside of that.  It made sense to just remove it, but if there is possible future use then it makes sense to keep it.

File: studio/components/ui/Usage.tsx [hunk 2]
```diff
@@ -1,12 +1,25 @@
...
 import { API_URL, PRICING_TIER_PRODUCT_IDS } from 'lib/constants'
-import { get, post } from 'lib/common/fetch'
+import { get } from 'lib/common/fetch'
 import SparkBar from 'components/ui/SparkBar'
 
+interface ApiUsageStats {
+  authUsers: string | null;
+  bucketSize: string | null;
+  dbSize: string | null;
+  dbTables: string | null;
```
[the remark this answers is not in the document]
Comment (nrayburn-tech) [reply studio/components/ui/Usage.tsx:14]: I am pretty sure that I checked this against the actual response I got from my supabase instance and it was returning null.  I don't know where the API for this endpoint exists, but if you can point me to that I can look at it.

I am having issues authenticating right now, so I can't check against my instance again right now.  Putting the URL I get here, but it is probably just some production issue that will be resolved shortly.  https://app.supabase.io/?error=server_error&error_description=failed+to+connect+to+`host=prod-2.c7y43uscdbtq.ap-southeast-1.rds.amazonaws.com+user=auth_admin+database=postgres`:+server+error+(FATAL:+remaining+connection+slots+are+reserved+for+non-replication+superuser+and+rds_superuser+connections+(SQLSTATE+53300



For our use case here, null and undefined are both being treated the same.  I could just make them all optional with string | null.  `authUsers?: string | null;` for example.

File: studio/components/to-be-cleaned/TooltipListener.tsx (hunk 1 above)
Comment (Isaiah-Hamilton) [reply studio/components/to-be-cleaned/TooltipListener.tsx:1]: For now lets keep it.

Comment (nrayburn-tech) [reply studio/components/to-be-cleaned/TooltipListener.tsx]: Reverted the commit.

**Comment by nrayburn-tech** [comment]
Updated and resolved the merge conflicts.

**Review by Isaiah-Hamilton — APPROVED** [review]
lgtm 🔥

**Comment by Isaiah-Hamilton** [comment]
@nrayburn-tech need to fix build error

image

**Comment by nrayburn-tech** [comment]
@Isaiah-Hamilton sorry it took me so long to get back to this, I didn't realize the build was failing.

It should be updated now.  Let me know if I need to do anything else.

**Pull request merged**